inference-engine/src/auto_plugin/auto_infer_request.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "auto_infer_request.hpp"
+#include <ie_input_info.hpp>
+#include <cpp_interfaces/interface/ie_iinfer_request_internal.hpp>
+
+namespace AutoPlugin {
+    using namespace InferenceEngine;
+
+AutoInferRequest::AutoInferRequest(const InputsDataMap&   networkInputs,
+                                   const OutputsDataMap&  networkOutputs,
+                                   const InferRequest&    inferRequest)
+    : IInferRequestInternal(networkInputs, networkOutputs)
+    , _inferRequest(inferRequest) {
+}
+
+std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> AutoInferRequest::GetPerformanceCounts() const {
+    return _inferRequest.GetPerformanceCounts();
+}
+
+void AutoInferRequest::InferImpl() {
+    _inferRequest.Infer();
+}
+
+void AutoInferRequest::SetBlob(const std::string& name, const InferenceEngine::Blob::Ptr& data) {
+    _inferRequest.SetBlob(name, data);
+}
+
+Blob::Ptr AutoInferRequest::GetBlob(const std::string& name) {
+    return _inferRequest.GetBlob(name);
+}
```

Comment:
we have a lot of other methods in InferRequest, e.g. Wait, StartAsync, etc. Theey are missed here

Do we need `AutoInferRequest`? I suppose we can return device infer request from AutoExecNetwork::CreateInferEquest()

Reply:
I think it may be impossible to return a `auto inferRequest = _network.CreateInferRequest();` directly in AutoExecNetwork, because `IInferRequestInternal` doesn't have such constructor API like `IInferRequestInternal(inferRequest)` and if implement it, it also needs define all the default functions which are in `AutoInferRequest` now. maybe it is not a good idea.